File: codes/web/src/types/api/views/project.ts
export type MemberAvatarPreview = {
  user_id: string;
  display_name: string;
  email: string;
  initial: string;
};

export type ProjectMembersSummary = {
  total: number;
  owner_count: number;
  member_count: number;
  owners_preview: MemberAvatarPreview[];
  members_preview: MemberAvatarPreview[];
};

export type ProjectScheduleStats = {
  task_total: number;
  todo_count: number;
  doing_count: number;
  done_count: number;
  archived_count: number;
  health_percent: number | null;
};

export type ProjectDashboardView = {
  workspace_id: string;
  workspace_name: string;
  project_id: string;
  name: string;
  description?: string | null;
  archived: boolean;
  can_manage: boolean;
  created_at?: string | null;
  created_at_label?: string | null;
  created_by_display_name?: string | null;
  members: ProjectMembersSummary;
  stats: ProjectScheduleStats;
};

File: codes/core-service/app/schemas/views/project.py
from datetime import datetime

from pydantic import BaseModel, Field

from app.schemas.views.schedule import ScheduleDashboardOut


class MemberAvatarPreviewOut(BaseModel):
    user_id: str
    display_name: str
    email: str
    initial: str


class ProjectMembersSummaryOut(BaseModel):
    total: int
    owner_count: int
    member_count: int
    owners_preview: list[MemberAvatarPreviewOut] = Field(default_factory=list)
    members_preview: list[MemberAvatarPreviewOut] = Field(default_factory=list)


class ProjectDashboardOut(BaseModel):
    workspace_id: str
    workspace_name: str
    project_id: str
    name: str
    description: str | None = None
    archived: bool
    can_manage: bool
    created_at: datetime | None = None
    created_at_label: str | None = None
    created_by_display_name: str | None = None
    members: ProjectMembersSummaryOut
    stats: ScheduleDashboardOut
